src/pages/index.js

```diff
@@ -15,80 +15,194 @@ import Section from "../components/Section";
-    },
-  },
-  selectors.cardSection
-);
-
 const newCardPopup = new PopupWithForm(
   selectors.newCardModal,
   handleAddCardSubmit
 );
 
-const editUserInfo = new UserInfo(profileSelectors);
-
 const editUserInfoPopup = new PopupWithForm(
   selectors.editProfileModal,
   handleProfileEditSubmit
 );
 
+const editAvatarPopup = new PopupWithForm(
+  selectors.editAvatarModal,
+  handleAvatarEditSubmit
+);
+
+const deleteCardPopup = new PopupConfirmDelete(selectors.confirmDeleteModal);
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "2f5abbfd-693d-4973-87a8-2e8fc711af51",
+    "Content-Type": "application/json",
+  },
+});
+
 //Initialize all instances
-cardSection.renderItems();
 imagePreviewPopup.setEventListeners();
 newCardPopup.setEventListeners();
 editUserInfoPopup.setEventListeners();
+editAvatarPopup.setEventListeners();
+deleteCardPopup.setEventListeners();
+
+let cardSection;
+let userInfo;
+
+api
+  .initialPageLoad()
+  .then(([userData, cards]) => {
+    cardSection = new Section(
+      {
+        items: cards,
+        renderer: (item) => {
+          cardSection.addItem(createCard(item));
+        },
+      },
+      selectors.cardSection
+    );
+    cardSection.renderItems();
+
+    userInfo = new UserInfo(profileSelectors);
+    userInfo.setUserInfo(userData.name, userData.about);
+    userInfo.setUserAvatar(userData.avatar);
+  })
+  .catch((err) => {
+    console.error(`There is an Error: `, err); // log the error to the console
+  });
 
 //Specific Event Listeners
 addNewCardButton.addEventListener("click", () => {
   formValidators["place-form"].toggleButtonState();
   newCardPopup.open();
 });
 
+let userData;
+
 editProfileButton.addEventListener("click", () => {
-  const userInfo = editUserInfo.getUserInfo();
+  userData = userInfo.getUserInfo();
   editUserInfoPopup.setInputValues({
-    title: userInfo.name,
-    description: userInfo.job,
+    title: userData.name,
+    description: userData.job,
   });
   formValidators["profile-form"].resetValidation();
   editUserInfoPopup.open();
 });
 
+editAvatarButton.addEventListener("click", () => {
+  userData = userInfo.getUserInfo();
+  editAvatarPopup.setInputValues({
+    link: userData.src,
+  });
+  formValidators["avatar-form"].toggleButtonState();
+  editAvatarPopup.open();
+});
+
 // Functions
 function handleImageClick(link, name) {
   imagePreviewPopup.open(link, name);
 }
 
 function handleAddCardSubmit(cardData) {
-  cardSection.addItem(
-    createCard({ name: cardData.title, link: cardData.link })
-  );
+  newCardPopup.setButtonText(true);
+  api
+    .addNewCard(cardData.title, cardData.link)
+    .then((data) => {
+      cardSection.addItem(createCard(data));
+    })
+    .catch((err) => {
+      console.error(`There is an Error: `, err); // log the error to the console
+    })
+    .finally(() => {
+      newCardPopup.setButtonText(false);
+    });
+
   newCardPopup.close();
```

Comment:
`  newCardPopup.close();
`

You need to close modals only in blocks `then` after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well


**Please, correct it everywhere**